test/Microsoft.AspNet.Routing.Tests/Template/TemplateParserTests.cs

```diff
@@ -634,10 +688,7 @@ public void InvalidTemplate_InvalidParameterNameWithOpenBracketThrows()
         {
             ExceptionAssert.Throws<ArgumentException>(
                 () => TemplateParser.Parse("{a}/{a{aa}/{z}"),
```

Comment:
for the cases like {p{{} (parameter name = p{)., the parameter parsing will parse this but parameter name validation will fail as these are not valid parameter names. the only place where parse will return success where these appear in constraint part of the parameter as it is not validated right away.